File: cellular_automata/cpp/cellular/cellular.cpp
#include "cellular.hpp"

namespace ca {

  State make_initial_state(size_t size)
  {
    auto state = ca::State(size, ca::EMPTY);
    state[size / 2] = ca::FILLED;

    return state;
  }

  State next(State state, Rule rule) {
    return State(state.size(), EMPTY);
  }

}

File: cellular_automata/python/cellular/cellular.py
Cell = str
EMPTY = " "
FILLED = "@"

Rule = int
State = str


def make_initial_state(size: int) -> State:
    out = [EMPTY for _ in range(size)]
    out[size // 2] = FILLED

    return "".join(out)


def next_state(state: State, rule: Rule) -> State:
    return EMPTY * len(state)
